lib/cog/chat/slack/formatter.ex

```diff
@@ -62,11 +62,26 @@ defmodule Cog.Chat.Slack.Formatter do
       nil ->
         message
       [{start, length}|_] ->
-        original = String.slice(message, start, length)
+        # Regex.run returns indices of a char list which causes weirdness when
+        # unicode is present. We have to convert with erlang's older binary to
+        # list conversion (ignoring unicode codepoints), do the replace, and
+        # then convert back with the same older conversion function for the
+        # indices to match up correctly. This is possibly a bug in Elixir's
+        # `Regex.run`.
```

Comment:
@kevsmith Yes, that's the reason. I'm not actually sure if there's the potentially to generate an invalid string here. I'll have to look at the implementation a bit more.

I think I remember the reason we don't `String.replace` being that there could be multiple identical matches in a string and we need to know which one to replace with with string. I might need to think about that a bit more, but your solution might still be ok.